File: Fleet.Api/Features/Ships/DTOs/OnShipLoadedEvent.cs
using Fleet.Api.Entities;

namespace Fleet.Api.Features.Ships.DTOs;

public class OnShipLoadedEvent
{
    public int ShipId { get; set; }
    public LoadedContainer LoadedContainer { get; set; }

    public OnShipLoadedEvent(int shipId, Container container)
    {
        ShipId = shipId;
        LoadedContainer = new LoadedContainer(container);
    } 
}

public class LoadedContainer
{
    public int Id { get; set; }
    public string Name { get; set; }
    
    
    public LoadedContainer(Container container)
    {
        Id = container.Id;
        Name = container.Name;
    } 
}

File: Fleet.Client/src/app/_models/onShipLoadedEvent.ts
import {Container} from "./container";

export interface OnShipLoadedEvent {
  shipId: number;
  loadedContainer: Container;
}

export function emptyShipLoadedEvent(): OnShipLoadedEvent{
  return { shipId: -1, loadedContainer: { id: -1, name: '' } }
}
